core/src/main/java/org/elasticsearch/index/mapper/fixed/FixedPointFieldMapper.java

```diff
@@ -0,0 +1,399 @@
+                        // would cause an overflow in decimalFactor later
+                        //
+                        // Long.MAX_VALUE ==  9223372036854775807
+                        // 10^19          == 10000000000000000000
+                        throw new MapperParsingException("Property [decimal_places] must be smaller than 19.");
+                    }
+
+                    builder.decimalPlaces((byte)decimalPlaces);
+                    iterator.remove();
+                }
+            }
+            return builder;
+        }
+    }
+
+    public static class FixedPointFieldType extends NumberFieldType {
+        private long decimalPlaces = Defaults.DECIMAL_PLACES;
+        private long decimalFactor = Defaults.DECIMAL_FACTOR;
+
+        public FixedPointFieldType() {
+            super(NumericType.LONG);
+        }
+
+        protected FixedPointFieldType(FixedPointFieldType ref) {
+            super(ref);
+        }
+
+        public void setDecimalPlaces(long decimalPlaces) {
+            this.decimalPlaces = decimalPlaces;
+            this.decimalFactor = (long) Math.pow(10.0, decimalPlaces);
+        }
+
+        public long getDecimalPlaces() {
+            return this.decimalPlaces;
+        }
+
+        public long getDecimalFactor() {
+            return this.decimalFactor;
+        }
+
+        @Override
+        public NumberFieldType clone() {
+            return new FixedPointFieldType(this);
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public Double nullValue() {
+            return (Double)super.nullValue();
+        }
+
+        @Override
+        public Long value(Object value) {
+            if (value == null) {
+                return null;
+            }
+            if (value instanceof Number) {
+                return ((Number) value).longValue();
+            }
+            if (value instanceof BytesRef) {
+                return Numbers.bytesToLong((BytesRef) value);
+            }
+            return Long.parseLong(value.toString());
+        }
+
+        @Override
+        public BytesRef indexedValueForSearch(Object value) {
+            long longValue = (long)(parseDoubleValue(value) * decimalFactor);
+            BytesRefBuilder bytesRef = new BytesRefBuilder();
+            NumericUtils.longToPrefixCoded(longValue, 0, bytesRef);   // 0 because of exact match
+            return bytesRef.get();
+        }
+
+        @Override
+        public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower, boolean includeUpper) {
+            return NumericRangeQuery.newLongRange(name(), numericPrecisionStep(),
+                lowerTerm == null ? null : (long) (parseDoubleValue(lowerTerm) * decimalFactor),
+                upperTerm == null ? null : (long) (parseDoubleValue(upperTerm) * decimalFactor),
+                includeLower, includeUpper);
```

Comment:
I think we need to be careful here. For instance if includeLower is true but parseDoubleValue rounds lowerTerm down, then we need to create a long range that has includeLower=false